### PR TITLE
Lint tsdb and tsdb/engine package

### PR DESCRIPTION
**Pull request by nuss-justin**
- [ ] CHANGELOG.md updated
- [X] Rebased/mergable
- [X] Tests pass
- [X] Sign [CLA](http://influxdb.com/community/cla.html) (if not already signed)

Updates #4098

### CODE REVIEW

**Comment by nuss-justin** [comment]
Rebased

File: tsdb/batcher.go [hunk 1]
```diff
@@ -109,6 +109,8 @@ func (b *PointBatcher) Start() {
 	}()
 }
 
+// Stop stopts the batching process. Stop waits for the batching routine
```
Comment (e-dard) [inline tsdb/batcher.go]: `stopts`

File: tsdb/functions.go [hunk 2]
```diff
@@ -187,11 +188,12 @@ func initializeReduceFunc(c *influxql.Call) (reduceFunc, error) {
 	}
 }
 
+// InitializeUnmarshaller the correspoding UnmarshalFunc for the given function call.
```
Comment (e-dard) [inline tsdb/functions.go]: `InitializeUnmarshaller the correspoding`

File: tsdb/functions.go [hunk 3]
```diff
@@ -274,11 +276,11 @@ func MapCount(input *MapInput) interface{} {
 	return n
 }
 
-type InterfaceValues []interface{}
+type interfaceValues []interface{}
```
Comment (e-dard) [inline tsdb/functions.go]: What's your rationale for unexporting this previously exported type?

File: tsdb/functions.go [hunk 4]
```diff
@@ -377,11 +379,12 @@ func ReduceCountDistinct(values []interface{}) interface{} {
 	return len(index)
 }
 
-type NumberType int8
```
Comment (e-dard) [inline tsdb/functions.go]: Ditto on the unexporting

File: tsdb/meta.go [hunk 5]
```diff
@@ -1125,12 +1126,6 @@ func (m *Measurement) DimensionTagSets(stmt *influxql.SelectStatement) ([]*influ
 	return tagSets, nil
 }
 
-type SelectInfo struct {
```
Comment (e-dard) [inline tsdb/meta.go]: @jwilder can this go? There are no references to it in the code-base.

**Comment by e-dard** [comment]
@nuss-justin can you sign the [CLA](https://influxdb.com/community/cla.html)?

File: tsdb/functions.go (hunk 3 above)
Comment (nuss-justin) [inline tsdb/functions.go]: The type is only used in the tsdb package and doesn't really look like something that should necessarily be exported, so I went by the suggestion to "favor making symbols private where possible before documenting.". 

Looking at the code more closely, I see that it's used inside the ReduceDistinct function and one could argue that it's needed to call ReduceDistinct with values that were not returned by another function from this package.

File: tsdb/functions.go (hunk 4 above)
Comment (nuss-justin) [inline tsdb/functions.go]: The type is only used for reducing (and only in this package) so, as with interfaceValues, I made it unexported (as I saw no use for it outside the package).

If you want I can just make it exported again and document it.

**Comment by nuss-justin** [comment]
@e-dard I signed the CLA (again?)

**Comment by jwilder** [comment]
I think #5196 might have removed a lot of code changed here.

**Comment by nuss-justin** [comment]
@jwilder Updated to current master :-)

**Comment by jwilder** [comment]
:+1: Thanks @nuss-justin

**Pull request merged**